p2p/protocol/identify/id.go

```diff
@@ -341,48 +347,26 @@ func (ids *idService) removeConn(c network.Conn) {
 	ids.connsMu.Unlock()
 }
 
-func (ids *idService) identifyConn(c network.Conn, signal chan struct{}) {
-	var (
-		s   network.Stream
-		err error
-	)
-
-	defer func() {
-		close(signal)
-
-		// emit the appropriate event.
-		if p := c.RemotePeer(); err == nil {
-			ids.emitters.evtPeerIdentificationCompleted.Emit(event.EvtPeerIdentificationCompleted{Peer: p})
-		} else {
-			ids.emitters.evtPeerIdentificationFailed.Emit(event.EvtPeerIdentificationFailed{Peer: p, Reason: err})
-		}
-	}()
-
-	s, err = c.NewStream(network.WithUseTransient(context.TODO(), "identify"))
+func (ids *idService) identifyConn(c network.Conn) error {
+	s, err := c.NewStream(network.WithUseTransient(context.TODO(), "identify"))
```

Comment:
shouldnt we have a timeout here? Or do we set the deadline?

Reply:
Let's do that in a separate PR. We do set a bunch of deadlines, but only read deadlines, which can lead to funny edge cases if the peer withholds flow control credit.